test/tests/functional/pbs_power_provisioning_cray.py

```diff
@@ -449,15 +500,536 @@ def test_last_used_time(self):
+            "/opt/cray/capmc/default/bin/capmc node_off", starttime=t)
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.expect(NODE, {'state': 'sleep'}, id=n['id'])
+        a = {"node_idle_limit": "1800", 'min_node_down_delay': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+
+        jid1 = self.submit_job(1000, {'Resource_List.vnode': self.names[0],
+                                      'Resource_List.vntype': 'cray_compute'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        t = int(time.time())
+        self.logger.info("Waiting for 10 mins to poweron the calandered nodes")
+        time.sleep(600)
+        self.server.log_match(
+            "/opt/cray/capmc/default/bin/capmc node_on", starttime=t)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(NODE, {'state': 'job-exclusive'}, id=self.names[0])
+        a = {'queue_type': 'execution',
+             'started': 'True',
+             'enabled': 'True',
+             'priority': '150'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq1')
+        t = int(time.time())
+        jid2 = self.submit_job(1000, {ATTR_queue: 'wq1',
+                                      'Resource_List.vntype': 'cray_compute'})
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid1)
+        self.scheduler.log_match("Job preempted by requeuing", starttime=t)
+        self.cleanup_power_on()
+
+    def test_last_used_time_node_sort_key(self):
+        """
+        Test last_used_time as node sort key.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {
+                            'job_history_enable': 'True'})
+        i = 0
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                if i > 1:
+                    self.server.manager(MGR_CMD_SET, NODE, {
+                        'state': 'offline'}, id=n['id'])
+                i += 1
+        a = {'node_sort_key': '"last_used_time LOW " ALL'}
+        self.scheduler.set_sched_config(a)
+        jid = self.submit_job(
+            1, {'Resource_List.select': '1:ncpus=1',
+                'Resource_List.place': 'excl'})
+        self.server.expect(JOB, {'job_state': 'F'}, id=jid, extend='x')
+        status = self.server.status(JOB, id=jid, extend='x')
+        exec_vnode = status[0]['exec_vnode']
+        node1 = exec_vnode.split(':')[0][1:]
+        jid = self.submit_job(
+            1, {'Resource_List.select': '1:ncpus=1',
+                'Resource_List.place': 'excl'})
+        self.server.expect(JOB, {'job_state': 'F'}, id=jid, extend='x')
+        jid = self.submit_job(
+            1, {'Resource_List.select': '1:ncpus=1',
+                'Resource_List.place': 'excl'})
+        self.server.expect(JOB, {'job_state': 'F'}, id=jid, extend='x')
```

Comment:
You submit 3 jobs, but only query the node from the first and the third.  What is the purpose of the second job?

Reply:
I have 2 jobs submitted first , So that last_used_time is set on 2 nodes and then i as i am using it as node_sort_key  to make sure 3rd job falls on the node where last_used_time is low.

Reply:
Consider adding this explanation as a comment.  If I was unsure of what was going on, others might as well.

Reply:
added explaination